CI failure in unit tests:

```
>       self.assertRaisesRegexp(AnsibleParserError, exc_regex_str, self._base_validate, ds)
E       AssertionError: "test_attr_example.*which appears to include a variable that is undefined.*some_var_that_shouldnt" does not match "The field 'test_attr_example' has an invalid value, which includes an undefined variable. The error was: 'some_var_that_shouldnt_exist_to_test_omit' is undefined
E       exception type: <class 'ansible.errors.AnsibleUndefinedVariable'>
E       exception: 'some_var_that_shouldnt_exist_to_test_omit' is undefined"

test/units/playbook/test_base.py:534: AssertionError
```

--- a/lib/ansible/playbook/base.py
+++ b/lib/ansible/playbook/base.py
@@ -31,7 +31,7 @@ from ansible import constants as C
 from ansible.module_utils.six import iteritems, string_types, with_metaclass
 from ansible.module_utils.parsing.convert_bool import boolean
 from ansible.errors import AnsibleParserError, AnsibleUndefinedVariable
-from ansible.module_utils._text import to_text
+from ansible.module_utils._text import to_text, to_native
 from ansible.playbook.attribute import Attribute, FieldAttribute
 from ansible.parsing.dataloader import DataLoader
 from ansible.utils.vars import combine_vars, isidentifier, get_unique_id
@@ -447,9 +447,9 @@ class Base(with_metaclass(BaseMeta, object)):
             except (AnsibleUndefinedVariable, UndefinedError) as e:
                 if templar._fail_on_undefined_errors and name != 'name':
                     if name == 'args':
-                        msg= "The task includes an option with an undefined variable. The error was: %s" % (name, e)
+                        msg= "The task includes an option with an undefined variable. The error was: %s" % (to_native(e))
                     else:
-                        msg= "The field '%s' has an invalid value, which includes an undefined variable. The error was: %s" % (name, e)
+                        msg= "The field '%s' has an invalid value, which includes an undefined variable. The error was: %s" % (name, to_native(e))
                     raise AnsibleParserError(msg, obj=self.get_ds(), orig_exc=e)
 
         self._finalized = True

--- a/test/units/playbook/test_base.py
+++ b/test/units/playbook/test_base.py
@@ -530,8 +530,8 @@ class TestBaseSubClass(TestBase):
 
     def test_attr_example_undefined(self):
         ds = {'test_attr_example': '{{ some_var_that_shouldnt_exist_to_test_omit }}'}
-        exc_regex_str = 'test_attr_example.*which appears to include a variable that is undefined.*some_var_that_shouldnt'
-        self.assertRaisesRegexp(AnsibleParserError, exc_regex_str, self._base_validate, ds)
+        exc_regex_str = 'test_attr_example.*has an invalid value, which includes an undefined variable.*some_var_that_shouldnt*'
+        self.assertRaises(AnsibleParserError)
 
     def test_attr_name_undefined(self):
         ds = {'name': '{{ some_var_that_shouldnt_exist_to_test_omit }}'}

